lib/dor/services/client/object.rb

```diff
@@ -148,6 +148,20 @@ def reindex
           raise_exception_based_on_response!(resp)
         end
 
+        # Verify the object can be indexed into Solr
+        # @raise [UnprocessableContent] when the response is not successful.
+        # @return [string] JSON representation of the descriptive solr document
+        def indexable(cocina:)
+          resp = connection.post do |req|
+            req.url "#{object_path}/indexable"
+            req.headers['Content-Type'] = 'application/json'
+            req.body = cocina.to_json
+          end
+          return true if resp.success?
+
+          raise_exception_based_on_response!(resp)
+        end
```

Comment:
Remind me, is the intent of this endpoint to check whether a specific, current Cocina object is indexable, or is the intent to be able to check *any* Cocina object is indexable?

Reply:
The intent is to check if the expected cocina data will be indexable, so this will need to be done BEFORE actual cocina is saved so that we do not raise an indexing error in the accessioning job. So it could be existing, or brand new, we cannot assume that the object actually exists, it is only determining (essentially) if `to_solr` does not raise an error on the provided descriptive metadata.

In short, this is verifying that a cocina representation is going to be indexable.

Reply:
@aaron-collier OK, then I wonder if the `ObjectsClient` (or elsewhere?) might be a better home for this code. The `ObjectClient` as I understand it now is for when you have a specific druid. To get an instance, you need to call `#object(DRUID_HERE)`.

Reply:
We decided to roll with this design until we ship it to Argo, at which point it may be clearer how this should all work.